doc/specs/#2634 - Broadcast Input/#2634 - Broadcast Input.md

```diff
@@ -0,0 +1,374 @@
+though.
+
+## Potential Issues
+
+<table>
+
+<tr>
+<td><strong>Compatibility</strong></td>
+<td>
+
+[comment]: # Will the proposed change break existing code/behaviors? If so, how, and is the breaking change "worth it"?
+
+</td>
+</tr>
+</table>
+
+[comment]: # If there are any other potential issues, make sure to include them here.
+
+### Future Considerations
+
+Let's look to iTerm2, who's supported this feature for years, for some
+inspiration of future things we should be worried about. If their users have
+asked for these features, then it's inevitable that our users will too 😉
+
+* [iterm2#6709]: Broadcast Input to multiple windows
+  - This is pretty straightforward. Would require coordination with the Monarch
+    though, and I'm worried about the perf hit of tossing every keystroke across
+    the process boundary.
+  - I suppose this would be `{ "action": "toggleBroadcastInput", "scope":
+    "global" }`
+* [iterm2#6451], [iterm2#5563]: "Broadcast commands"
+  - iTerm2 has an action that lets the user manually clear the terminal-side
+    buffer. (This is tracked on the Windows Terminal as [#1882]). It might make
+    sense for there to be a mode where some _actions_ are also broadcast to
+    panes, not just key strokes. But which actions would those be? Moving the
+    selection anchors? Copy doesn't really make sense. Paste _does_ though.
+    Maybe the open find dialog / next&prev search match actions?
+  - This probably would require it's own spec.
+* [iterm2#6007]: Different stripe color for different broadcast modes
+  - Have one color to indicate when broadcasting in `global` scope, another in
+    `tab` scope, a third in `pane` scope.
+  - This might mesh well with theming ([#3327]), for properties like
+    `pane.broadcastBorderColor.globalScope`,
+    `pane.broadcastBorderColor.paneScope`. Don't love those names, but you get
+    the idea.
+* **[iterm2#5639]: Broadcast groups**, [iterm2#3372]: Broadcast Input to
+  multiple but not all tabs
+  - This is probably the most interesting request. I think this one identifies a
+    major shortcoming of the above proposals. With proposal 2, there's only ever
+    one top-level broadcast group. With proposals 1 & 3, there's per-tab
+    broadcast groups. In neither proposal can you have multiple concurrent
+    side-by-side broadcast groups.
+  - Groups should probably work across tabs. This would suggest that Proposal 2
+    is closer to how groups would work. Instead of there being one top-level
+    set, there would be multiple. **I'm not sure how proposals 1&3 would
+    seamlessly transition into also supporting groups**.
+  - The major trick here is: how do we differentiate these different groups to
+    the user? If we used the broadcast icon with a number, maybe in the corner
+    of the tab? Like [📡: 1]? Can a pane be in multiple broadcast sets at the
+    same time?
+  - The natural arg idea would be `{ "action": "toggleBroadcastInput", "scope":

```

Comment:
Oh maybe we want something like `group: -1` or something that indicates "create a new group". Cause the initial implementation is going to be "add all the panes in this tab to a new group", and each tab has it's own group. So the action added in #9222 would default to "make a new group".

But the default value for the arg shouldn't be `-1`, because adding each pane to its own broadcast group is useless